Classes/Notifications/NotificationSubscriptionsController.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  NotificationSubscriptionsController.swift
+//  Freetime
+//
+//  Created by Ryan Nystrom on 12/18/17.
+//  Copyright © 2017 Ryan Nystrom. All rights reserved.
+//
+
+import UIKit
+
+final class NotificationSubscriptionsController {
+
+    enum State {
+        case loading
+        case results([Repository])
+        case error
+    }
+
+    private weak var viewController: UIViewController?
+    private let client: NotificationClient
+    private var state: State = .loading
+
+    init(viewController: UIViewController, client: NotificationClient) {
+        self.viewController = viewController
+        self.client = client
+    }
+
+    // MARK: Public API
+
+    var actions: [UIAlertAction] {
+        switch state {
+        case .loading: return []
+        case .error: return [refreshAction]
+        case .results(let repos): return repoActions(repos: repos)
+        }
+    }
+
+    public func fetchSubscriptions() {
+        client.fetchWatchedRepositories { [weak self] result in
+            switch result {
+            case .success(let repos): self?.state = .results(repos)
+            case .error:
+                self?.state = .error
+                ToastManager.showGenericError()
+            }
+        }
+    }
+
+    // MARK: Private API
+
+    var refreshAction: UIAlertAction {
+        return UIAlertAction(
+            title: NSLocalizedString("Refresh", comment: ""),
```

Comment:
Wondering where this is used - why would we "hide" a refresh in a `UIAlertAction` if anywhere else we use a `UIRefreshControl`?

Reply:
So you should almost never see this. I'm making sure that if the fetch request to get your watched repos fails, there's at least a button to refresh it. Maybe it should say something like "Error - Refresh"? Idk, there's such limited space here.